cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/transaction/TransactionRunners.java

```diff
@@ -0,0 +1,254 @@
+   * @throws X1 if failed to execute the given {@link TxRunnable} in a transaction. If the
+   * TransactionException has a cause in it, the cause is thrown as-is if it is an instance of X1.
+   * @throws X2 if failed to execute the given {@link TxRunnable} in a transaction. If the
+   * TransactionException has a cause in it, the cause is thrown as-is if it is an instance of X2.
+   * @throws RuntimeException if cause is not an instance of X1 or X2. The cause is wrapped with
+   * {@link RuntimeException} if it is not already a {@link RuntimeException}.
+   */
+  public static <X1 extends Throwable, X2 extends Throwable> void run(TransactionRunner txRunner,
+                                                                      TxRunnable runnable,
+                                                                      Class<X1> exception1,
+                                                                      Class<X2> exception2) throws X1, X2 {
+    try {
+      txRunner.run(runnable);
+    } catch (TransactionException e) {
+      throw propagate(e, exception1, exception2);
+    }
+  }
+
+  /**
+   * Executes the given {@link TxCallable} using the given {@link TransactionRunner}.
+   * @param txRunner the {@link TransactionRunner} to use for txRunner execution.
+   * @param callable the {@link TxCallable} to be executed inside a transaction
+   * @param <V> type of the result
+   * @return value returned by the given {@link TxCallable}.
+   * @throws  RuntimeException if failed to execute the given {@link TxRunnable} in a transaction.
+   * If the TransactionException has a cause in it, the cause is propagated.
+   */
+  public static <V> V run(TransactionRunner txRunner, TxCallable<V> callable) {
+    AtomicReference<V> result = new AtomicReference<>();
+    try {
+      txRunner.run(context -> result.set(callable.call(context)));
+    } catch (TransactionException e) {
+      throw propagate(e);
+    }
+    return result.get();
+  }
+
+  /**
+   * Executes the given {@link TxCallable} using the given {@link TransactionRunner}.
+   *
+   * @param txRunner the {@link TransactionRunner} to use for the transaction execution.
+   * @param callable the {@link TxCallable} to be executed inside a transaction
+   * @param <V> type of the result
+   * @param <X> exception type of propagate type
+   * @return value returned by the given {@link TxCallable}.
+   * @throws X if failed to execute the given {@link TxRunnable} in a transaction. If the TransactionException
+   * has a cause in it, the cause is thrown as-is if it is an instance of X.
+   * @throws RuntimeException if cause is not an instance of X. The cause is wrapped with {@link RuntimeException}
+   * if it is not already a {@link RuntimeException}.
+   */
+  public static <V, X extends Throwable> V run(TransactionRunner txRunner,
+                                               TxCallable<V> callable, Class<X> exception) throws X {
+    AtomicReference<V> result = new AtomicReference<>();
```

Comment:
It's better to have variable declare in the closest scope that is using it. It this case, it is better to write it as:

```
try {
  AtomicReference<V> result = new AtomicReference<>();
  txRunner.run(c -> result.set(callable.call(c)));
  return result.get();
} catch (TransactionException e) {
  throw propagate(e, exception);
}
```

This is also more robust against future changes in the `catch` block